ui/plugins/controller.js

```diff
@@ -334,6 +334,29 @@ treeherder.controller('PluginCtrl', [
             }
         };
 
+        $scope.retriggerJobs = function(jobs) {
+            if($scope.user.loggedin) {
+                if(jobs.length > 5) {
+                    if (!window.confirm('This will trigger ' + jobs.length + ' jobs! Are you sure?')) {
+                        return;
+                    }
+                }
+
+                // Initial promise for sequential execution.
+                var jobPromise = Promise.resolve();
+                jobs.forEach(function(job) {
+                    jobPromise = jobPromise.then(function() {
+                        return selectJob(job.id);
+                    });
```

Comment:
I think the best way to deal with this is to convert the retrigger endpoint to a list_route and pass it a list of jobs. I prefer a single job being a special case rather than make potentially >1000 requests to the api as a consequence of a click.